### PR TITLE
Remove prose test from retryable writes specification

### PR DESCRIPTION
**Pull request by jstewart-mongo**
Remove the prose test from the retryable writes test specification
https://github.com/mongodb/specifications/tree/master/source/retryable-writes/tests#replica-set-failover-test

### CODE REVIEW

File: source/retryable-writes/tests/README.rst [hunk 1]
```diff
@@ -210,7 +210,7 @@ testing an update or delete that is split into two commands, the ``skip`` should
 be set to the number of statements in the first command to allow the fail point
 to trigger on the second command.
 
-Replica Set Failover Test
+Replica Set Failover Test (TO BE REMOVED)
```
Comment (ShaneHarvey) [inline source/retryable-writes/tests/README.rst]: Can you remove this entire section?

Comment (jstewart-mongo) [reply source/retryable-writes/tests/README.rst]: Done

**Comment by jmikola** [comment]
@jstewart-mongo: Please add a reference to [SPEC-1200](https://jira.mongodb.org/browse/SPEC-1200) to the commit before merging.

**Comment by mbroadst** [comment]
@ShaneHarvey are we good to go here? @jstewart-mongo please merge once Shane has approved.

**Pull request closed without merging**